### ISSUE TITLE
Use Events List data from an API and confine search to that List

### ISSUE THREAD

**Issue by Fnu-Avi** [SharePoint/sp-dev-solutions#350]
### Category
- [x] Question
- [ ] Bug
- [ ] Enhancement

I recently started with SPFX and need to use this search web part to only search from an api (which is Events List API). How can I specify my API URL and where should I go for that.

I just need to limit my search functionalities to that API. I would appreciate any leads onto this.

Thanks for your contribution! Sharing is caring.

**Comment by wobba** [comment]
If you are talking about the modern search web part, it only supports searching SharePoint content.

**Comment by Fnu-Avi** [comment]
Currently its searching within my whole sharepoint environment. 
Ex: 
Currently, its searching in the whole abc.sharepoint.com for the following link.
**link**: https://abc.sharepoint.com/sites/sub-abc-Intranet 

I've many subsites within this domain.
I want it to only search in the sub site: "sub-abc-Intranet". Also, if this is feasible I want to perform the same search functionality on a particular list in a sub site. 

I want to perform search functionality based on the subsite level and list level.

**Comment by wobba** [comment]
Use the query:
`path:https://abc.sharepoint.com/sites/sub-abc-Intranet` to limit to items starting with that path (one approach at least). I recommend reading up on KQL and search queries in the official docs or at my blog over at https://www.techmikael.com/2014/03/s15e01-kql-basics.html

**Comment by Fnu-Avi** [comment]
I got the data from the list. Now, I've that data and it's been displayed. 

The issue which I'm facing now is to use that result which I got from using path:https://abc.sharepoint.com/sites/sub-abc-Intranet to another search box as an input. 

Ex:
I've a search box and I used KQL query to get the data from my sub-site.
Now, I need to perform the search operations on only those results.

Help Appreciated.
Thanks @wobba
